src/gar.lua

```diff
@@ -369,18 +369,40 @@ function gar.decreaseDelegateStake(gatewayAddress, delegator, qty, currentTimest
 		error("Remaining delegated stake must be greater than the minimum delegated stake amount.")
 	end
 
-	-- Withdraw the delegate's stake
+	-- Instant withdrawal logic with penalty
+	if instantWithdraw then
+		-- Unlock the tokens from the gateway and delegate
+		gateway.delegates[delegator].delegatedStake = gateway.delegates[delegator].delegatedStake - qty
+		gateway.totalDelegatedStake = gateway.totalDelegatedStake - qty
 
-	local newDelegateVault = {
-		balance = qty,
-		startTimestamp = currentTimestamp,
-		endTimestamp = currentTimestamp + gar.getSettings().delegates.withdrawLengthMs,
-	}
+		-- Calculate the penalty amount
+		local maxPenalty = 0.80
```

Comment:
nit: use the constant variable you created

Reply:
https://github.com/ar-io/ar-io-network-process/pull/97/commits/0f8b37a3d29046fead347d62fe2cf4f9f146c711